pkg/declarativeconfig/transform/notifier.go

```diff
@@ -0,0 +1,116 @@
+package transform
+
+import (
+	"reflect"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/declarativeconfig"
+	"github.com/stackrox/rox/pkg/errox"
+)
+
+var (
+	_ Transformer = (*notifierTransform)(nil)
+
+	notifierType = reflect.TypeOf((*storage.Notifier)(nil))
+)
+
+type notifierTransform struct{}
+
+func newNotifierTransform() *notifierTransform {
+	return &notifierTransform{}
+}
+
+func (r *notifierTransform) Transform(configuration declarativeconfig.Configuration) (map[reflect.Type][]proto.Message, error) {
+	notifierConfig, ok := configuration.(*declarativeconfig.Notifier)
+	if !ok {
+		return nil, errox.InvalidArgs.Newf("invalid configuration type received for notifier: %T", configuration)
+	}
+
+	if notifierConfig.Name == "" {
+		return nil, errox.InvalidArgs.CausedBy("name must be non-empty")
+	}
+
+	notifierTypeStr, err := getNotifierType(notifierConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid notifier type")
+	}
+
+	notifierProto := &storage.Notifier{
+		Id:     declarativeconfig.NewDeclarativeNotifierUUID(notifierConfig.Name).String(),
+		Name:   notifierConfig.Name,
+		Type:   notifierTypeStr,
+		Traits: &storage.Traits{Origin: storage.Traits_DECLARATIVE},
+	}
+
+	if notifierConfig.GenericConfig != nil {
+		notifierProto.Config = getGenericConfig(notifierConfig.GenericConfig)
+	} else if notifierConfig.SplunkConfig != nil {
+		notifierProto.Config = getSplunkConfig(notifierConfig.SplunkConfig)
+	} else {
+		return nil, errox.InvalidArgs.Newf("unsupported notifier type %s", notifierTypeStr)
+	}
```

Comment:
nit: could make this a separate function?

Also, it might be worthwhile to already make this a `switch` in the case that we are going to extend this with more configuration types.

Reply:
The reason why it's not possible to make it into a separate function is that the umbrella type for `GenericConfig` and `SplunkConfig` is private `isNotifier_Config` type, so it can't be used in the function. I also decided against making a non-pure function